nibabel/tests/test_scripts.py

```diff
@@ -54,15 +54,62 @@ def script_test(func):
 
 
 @script_test
-def test_nib_ls():
+def _test_nib_ls_example4d(opts=[], hdrs_str=""):
     # test nib-ls script
     fname = pjoin(DATA_PATH, 'example4d.nii.gz')
     expected_re = (" (int16|[<>]i2) \[128,  96,  24,   2\] "
-                   "2.00x2.00x2.20x2000.00  #exts: 2 sform$")
-    cmd = ['nib-ls', fname]
+                   "2.00x2.00x2.20x2000.00  #exts: 2%s sform$"
+                   % hdrs_str)
+    cmd = ['nib-ls'] + opts + [fname]
     code, stdout, stderr = run_command(cmd)
     assert_equal(fname, stdout[:len(fname)])
-    assert_not_equal(re.match(expected_re, stdout[len(fname):]), None)
+    assert_re_in(expected_re, stdout[len(fname):])
+
+@script_test
+def test_nib_ls():
+    yield _test_nib_ls_example4d
```

Comment:
Don't feel too strongly, but I prefer to avoid yield because it makes the test failures harder to debug.